packages/graphql-testing/README.md

```diff
@@ -0,0 +1,94 @@
+# `@shopify/graphql-testing`
+
+[![Build Status](https://travis-ci.org/Shopify/quilt.svg?branch=master)](https://travis-ci.org/Shopify/quilt)
+[![License: MIT](https://img.shields.io/badge/License-MIT-green.svg)](LICENSE.md) [![npm version](https://badge.fury.io/js/%40shopify%2Fgraphql-testing.svg)](https://badge.fury.io/js/%40shopify%2Fgraphql-testing.svg)
+
+This utility aid with the following testing scenarios:
+
+1. Testing graphql request with mock data.
+2. Testing the state of your application before / after graphql request(s) resolves.
+
+It does this by creating a mock `ApolloClient` that can be access using `const graphQL = createGraphQLFactory()(mockData); const apolloClient = graphQL.client` and pass into your test setup.
+
+To assert with operations that had been executed. Find list of operations using `apolloClient.operations.all()`.
+
+To resolve all pending graphql request(s). Use `await graphQL.resolveAll()` in your test.
+
+## Installation
+
+```bash
+$ yarn add @shopify/graphql-testing
+```
+
+## Usage
+
+The following is an example of how to use `createGraphQLFactory` in a react component test.
+
+Typically you will want to generalized some of those implementation (ie. mouting of `ApolloProvider`) for repeated use.
```

Comment:
Note: In a typical application you will want to generalize parts of this implementation (ie. mounting the `ApolloProvider`) for repeated use.